mypyc/lib-rt/librt_internal.c

```diff
@@ -499,9 +534,13 @@ write_float(PyObject *self, PyObject *const *args, size_t nargs, PyObject *kwnam
 
 /*
 int format:
-    most common values (-10 <= value <= 117): single byte as `(uint8_t)(value + 10) << 1`
-    medium values (fit in CPyTagged): \x01 followed by CPyTagged value
-    long values (very rare): \x03 followed by decimal string (see str format)
+    one byte: last bit 0, 7 bits used
+    two bytes: last two bits 01, 14 bits used
+    four bytes: last three bits 011, 29 bits used
+    everything else: 00000111 followed by serialized string representation
```

Comment:
What about using ...01111 (four 1 bits) for the final option, as this would leave ...0111 available for a possible 60-bit format in the future. (I don't have a strong opinion.)

Reply:
OK. TBH I doubt we will ever need this, but it is very easy to make this change now.